custom_components/srf_weather/weather.py

```diff
@@ -104,36 +104,38 @@ async def get_api_key(hass: HomeAssistantType, data: MutableMapping) -> str:
     return data[ATTR_API_KEY]
 
 
-async def _get(hass, api_data: dict, url: str, **kwargs) -> dict:
+async def _get(hass, api_data: dict, url: str, expected_mime_type: str, **kwargs) -> dict:
     session = async_get_clientsession(hass)
     api_key = await get_api_key(hass, api_data)
     weak_update(
         kwargs,
         "headers",
         {
             "Authorization": f"Bearer {api_key}",
+            "Content-Type": "application/json"
         },
     )
     logger.debug("GET %s with %s", url, kwargs)
     async with session.get(url, **kwargs) as resp:
-        if resp.status == HTTP_OK:
-            logger.debug(
-                "Rate-limit available %s, rate-limit reset will be on %s",
-                resp.headers.get("x-ratelimit-available"),
-                datetime.fromtimestamp(
-                    int(resp.headers.get("x-ratelimit-reset-time", 0)) / 1000
-                ),
-            )
-        data = await resp.json()
-        logger.debug("response: %s", data)
+        logger.debug("API Response: %s", resp)
         resp.raise_for_status()
 
+        logger.debug(
+            "Rate-limit available %s, rate-limit reset will be on %s",
+            resp.headers.get("x-ratelimit-available"),
+            datetime.fromtimestamp(
+                int(resp.headers.get("x-ratelimit-reset-time", 0)) / 1000
+            ),
+        )
+
+        data = await resp.json(content_type=expected_mime_type)
```

Comment:
Ah I see you already did that once. Is there a good reason why you removed it again?
If you have to pass in "text/html" you might as well ignore it because at that point content type is meaningless.

Reply:
Well, in theory content type isn't meaningless. While it is at the moment, some future format might really need another parser.... I was thinking I'd rather prefer to *know* when they change content type. But then, its a mess anyways, so I don't care that much :man_shrugging:

Reply:
We'll know because it'll break in other ways :D
Since we're not handling the exception at the moment, it really doesn't matter whether it's a JSON decode error or the content type mismatch error.

Reply:
When you know the content type mismatch, you already know your if statement :)

Reply:
But yeah, that is the trade off, it might also just be JSON again, just with another content type... So ignoring would make it continue to work.

Reply:
Is my understanding correct then that you're not willing to make this change?

Side note:
I would hope that at some point someone working on the SRF API team is gonna notice that the response's content type is blatantly incorrect and they're gonna change it back to json.

Reply:
Sorry, was busy with other stuff, and my API key had too many requests, so wasn't able to test anymore.

Ignoring the mime type now.

> I would hope that at some point someone working on the SRF API team is gonna notice that the response's content type is blatantly incorrect and they're gonna change it back to json.

They might not want to do that for compatibility reason... If they do, we are ready now :smile: